sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADHandleConditionalAccessFilter.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.aad.webapp;
+
+import com.azure.spring.aad.webapi.AADOAuth2OboAuthorizedClientRepository;
+import com.azure.spring.autoconfigure.aad.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Handle the {@link WebClientResponseException} in On-Behalf-Of flow.
+ *
+ * <p>
+ * When the Web API needs re-acquire token(The request requires higher privileges than provided by the access token in
+ * On-Behalf-Of flow.), it can sent a 403 with information in the WWW-Authenticate header to web client ,web client
+ * will throw {@link WebClientResponseException}, Web APP can handle this exception to challenge the user.
+ */
+public class AADHandleConditionalAccessFilter extends OncePerRequestFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADOAuth2OboAuthorizedClientRepository.class);
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain filterChain) throws IOException, ServletException {
+        try {
+            filterChain.doFilter(request, response);
+        } catch (Exception exception) {
+            WebClientResponseException webClientResponseException =
+                Optional.of(exception)
+                        .map(Throwable::getCause)
+                        .filter(e -> e instanceof WebClientResponseException)
+                        .map(e -> (WebClientResponseException) e)
+                        .filter(AADHandleConditionalAccessFilter::isConditionalAccessExceptionFromObo)
+                        .orElse(null);
+            if (webClientResponseException != null) {
+                handleConditionalAccess(webClientResponseException, request, response);
+                return;
+            }
+            throw exception;
+        }
+    }
+
+    private static boolean isConditionalAccessExceptionFromObo(WebClientResponseException exception) {
+        String result = Optional.of(exception)
+                                .map(WebClientResponseException::getHeaders)
+                                .map(httpHeaders -> httpHeaders.get(HttpHeaders.WWW_AUTHENTICATE))
```

Comment:
The above 2 lines of code execute 2 times, could you please remove the duplicate?

For example:
```
conditionalAccessClaims = ...;
if (conditionalAccessClaims != null ) {
  handleConditionalAccess(...);
}
```

Reply:
Good idea, I will change the code to execute one time, thinks.